src/components/organisms/Badges/Badges.tsx

```diff
@@ -0,0 +1,167 @@
+import { useSelector } from "hooks/useSelector";
+
+import { PLAYA_VENUE_NAME } from "settings";
+
+import { CampRoomData } from "types/CampRoomData";
+import { isVenueWithRooms } from "types/CampVenue";
+import { AnyVenue } from "types/Firestore";
+import { Venue } from "types/Venue";
+
+import { WithId } from "utils/id";
+import {
+  orderedVenuesSelector,
+  userModalVisitsSelector,
+} from "utils/selectors";
+import { venueInsideUrl, venuePreviewUrl } from "utils/url";
+import { notEmpty } from "utils/types";
+
+export const Badges: React.FC<{
+  user: WithId<User>;
+  currentVenue: WithId<Venue>;
+}> = ({ user, currentVenue }) => {
+  useFirestoreConnect([
+    {
+      collection: "users",
+      doc: user.id,
+      subcollections: [{ collection: "visits" }],
+      storeAs: "userModalVisits",
+    },
+  ]);
+  const visits = useSelector(userModalVisitsSelector);
+  const venues = useSelector(orderedVenuesSelector);
+
+  const { relatedVenues } = useConnectRelatedVenues({
+    venueId: currentVenue.id,
+    withEvents: false,
+  });
+  const relatedVenueIds = relatedVenues.map((venue) => venue.id);
+
+  const visitHours = useMemo(() => {
+    if (!visits) return 0;
+
+    const visitSeconds = visits
+      .filter((v) => v.id === currentVenue.id || relatedVenueIds.includes(v.id))
+      .reduce((acc, v) => acc + v.timeSpent, 0);
+    return Math.floor(visitSeconds / (60 * 60));
+  }, [visits, currentVenue, relatedVenueIds]);
+
+  // Only show visits to related venues
+  const relevantVisits = visits?.filter(
+    (visit) =>
+      currentVenue.id === visit.id || relatedVenueIds.includes(visit.id)
+  );
+
+  const badges = useMemo(() => {
+    if (!relevantVisits || !venues) return [];
+    return relevantVisits
+      .map((visit) => {
+        const { venue, room } = findVenueAndRoomByName(visit.id, venues);
+        if (
+          !venue ||
+          (currentVenue.id !== venue.id && !relatedVenueIds.includes(venue.id))
+        )
+          return undefined;
+
+        if (room) {
+          return {
+            venue,
+            room,
+            image: room.image_url,
+            label: room.title,
+          };
+        }
+
+        return {
+          venue,
+          image: venue.mapIconImageUrl,
+          label: venue.name,
+        };
+      })
+      .filter((b) => b !== undefined);
+  }, [relevantVisits, venues, currentVenue, relatedVenueIds]);
+
+  const badgeList = useMemo(
+    () =>
+      badges.filter(notEmpty).map((b) => (
+        <li className="badge-list-item" key={b.label}>
+          <Link to={getLocationLink(b.venue, b.room)}>
+            <img
+              className="badge-list-item-image"
+              src={b.image}
+              alt={`${b.label} badge`}
+            />
+          </Link>
+        </li>
+      )),
+    [badges]
+  );
+
+  if (!relevantVisits) {
+    return <>Visit venues to collect badges!</>;
+  }
+
+  return (
+    <>
```

Comment:
[nit] unneeded `React.Fragment` wrapper as we only have a single 'root `div`' here anyway.

Reply:
Hm, I tried to take away the `Fragment` and turns out we have 2 at the top level of this component: `<div className="visits"` and `<div className="badges-container"` so it does need to stay. There's an autofix linter in eslint that'll detect these, so we can & should use that when we're ready.